finance-service-api/src/main/kotlin/com/ajaxproject/financeservice/beanpostprocessor/NatsControllerBeanPostProcessor.kt

```diff
@@ -0,0 +1,38 @@
+package com.ajaxproject.financeservice.beanpostprocessor
+
+import com.ajaxproject.financeservice.controller.NatsController
+import com.google.protobuf.GeneratedMessageV3
+import io.nats.client.Connection
+import io.nats.client.Dispatcher
+import io.nats.client.Message
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+
+@Component
+class NatsControllerBeanPostProcessor(private val connection: Connection) : BeanPostProcessor {
```

Comment:
please decide which connection are you using, autowired in NatsControllerBeanPostProcessor or from NatsController. (from NatsControllerBeanPostProcessor is recommended)